GroupListView (need to come back for progressbar. currently set to 0

jiyoon-wd opened the pull request:
# Pull Request Title

## Description

What does this PR fix? You can mention the issue number like
so to reference the issue automatically: "Fixes #1"

## Screenshot/Recording (if applicable)

<img width="354" alt="Screenshot 2024-10-22 at 6 26 40 PM" src="https://github.com/user-attachments/assets/1cbea3e6-ee74-4103-976e-bc7e77cf73eb">

jiyoon-wd committed:
GroupListView (need to come back for progressbar. currently set to 0

jiyoon-wd committed:
modified GroupListView according to Figma and removed old GroupsListView

jiyoon-wd commented:
<img width="469" alt="Screenshot 2024-10-29 at 6 53 54 PM" src="https://github.com/user-attachments/assets/2a9c152c-ce35-4567-9900-29c916999cc1">

jiyoon-wd committed:
ZStack background to Vstack

jiyoon-wd committed:
changed colors of the title and buttons

jiyoon-wd committed:
Fixed merge conflict and changed the code so that we are fetching groups array of String (group.ids) not groups array of Object(Group) according to new model structure

jiyoon-wd committed:
^

jiyoon-wd committed:
updates

changed Group sample struct to static constant so that group is initializing only once - it was initializing with new unique id every time it was called (unable to get the right group)

jiyoon-wd commented:
<img width="469" alt="Screenshot 2024-10-30 at 1 48 07 AM" src="https://github.com/user-attachments/assets/ab599f9b-118e-4cca-aafb-634b36ff3404">

jiyoon-wd committed:
Merge branch 'main' into GroupListViewUI

jiyoon-wd committed:
Merge branch 'main' into GroupListViewUI

rahulon12 commented on Vineyard/Vineyard/GroupCardView.swift:
```diff
@@ -0,0 +1,55 @@
+            HStack {
+                Spacer()
+                ProgressView(value: Double(group.currScore), total:Double(group.scoreGoal))
+                    .progressViewStyle(LinearProgressViewStyle())
```
nit:
```suggestion
                    .progressViewStyle(.linear)
```

rahulon12 commented on Vineyard/Vineyard/GroupListView.swift:
```diff
@@ -0,0 +1,74 @@
+            }.ignoresSafeArea()
+        }
+        .sheet(isPresented: $isPresentingAddGroup) {
+            AddGroupView(isPresented: $isPresentingAddGroup, viewModel: viewModel)
```
Does this use Yash's new pipeline implemented in #41?

jiyoon-wd replied:
ye

https://github.com/user-attachments/assets/3f7da98e-c970-4825-8d0b-a956c88dc584

rahulon12 replied:
No, there's a new pipeline that Yash has. Pull from main and see

jiyoon-wd replied:
sorry I saw the video and didn't realize it is differentt
I changed so that it is using CreateGroupView rather than AddGroupView

https://github.com/user-attachments/assets/e1aa9765-ab73-4fa6-a84c-2e66da9d89f5

rahulon12 commented on Vineyard/Vineyard/GroupListViewModel.swift:
Do we need this file?

jiyoon-wd replied:
Naur ma bad

jiyoon-wd committed:
Updates

fixed minor issues before merging to main

rahulon12 commented on Vineyard/Vineyard/AddGroupView.swift:
I don't think you need this. Instead use the new pipeline from #41.

jiyoon-wd replied:
I now deleted AddGroupView as it is no longer needed.

jiyoon-wd committed:
Update for merging

changed to new pipeline, using CreateGroupView not AddGroupView
and deleted AddGroupView

rahulon12 commented on Vineyard/Vineyard/Group.swift:
```diff
@@ -14,14 +14,19 @@ class Group: Identifiable, Codable {
         self.resolutions = resolutions
         self.deadline = deadline
+        self.scoreGoal = scoreGoal;
+        self.currScore = 3
```
Just curious, any reason for this to be 3, not 0?

jiyoon-wd replied:
:D wanted to just test if the progress bar was working

rahulon12 reviewed (approved):
Looks good! Just a small question :)
